### DISCUSSION TITLE
authenticationKeyManager=null

### DISCUSSION THREAD

**Discussion by jijkbird** [serverpod/serverpod#679] [Q&A] [answered]
I am trying to write user-related logic code. The server sends the key to the client, but the client.authenticationKeyManager of the client is empty. What should I do

**Comment by jijkbird** [comment]
The function 'FlutterAuthenticationKeyManager' isn't defined.

**Comment by jijkbird** [comment]
I used serverpod_ auth_ shared_ Flutter, but the server session. auth. authenticatedUserId=null

**Comment by jijkbird** [comment]
client = Client('http://10.0.2.2:8080/',
      authenticationKeyManager: FlutterAuthenticationKeyManager())
    ..connectivityMonitor = FlutterConnectivityMonitor();
  await sessionManager.initialize();

**Comment by jijkbird** [comment]
print(await session.auth.authenticatedUserId);

**Comment by jijkbird** [comment]
print(await session.auth.authenticatedUserId);

**Comment by jijkbird** [comment]
I don't use oauto

**Comment by vlidholt** [comment, marked as answer]
On the server-side `await session.auth.authenticatedUserId` will be null until a user has signed in. You can sign in a user with email or social logins. Check out the auth module for more information on how to do this:
https://docs.serverpod.dev/concepts/authentication
